pgmpy/identification/instrument.py

```
@@ -0,0 +1,148 @@
+                    break
+
+        return scaling_indicators
+
+    def _iv_transformations(self, X, Y, causal_graph, scaling_indicators=None):
+        full_graph_ = causal_graph.copy()
+
+        exposures = full_graph_.get_role("exposures")
+        latent_variables = full_graph_.get_role("latents")
+        observed = full_graph_.get_role("observed")
+        scaling_indicators = self._get_scaling_indicators(causal_graph)
+
+        if not full_graph_.has_edge(X, Y):
+            raise ValueError(f"The edge from {X} -> {Y} does not exist in the graph")
+
+        if (X in exposures) and (Y in observed):
+            if full_graph_.has_edge(X, Y):
+                full_graph_.remove_edge(X, Y)
+            dependent_var = Y
+
+        elif Y in latent_variables:
+            full_graph_.add_edge(Y, scaling_indicators[Y])
+            dependent_var = scaling_indicators[Y]
+
+        else:
+            dependent_var = Y
+
+        variable_parents = [
+            var for var in causal_graph.predecessors(Y) if not var.startswith(".")
+        ]
+
+        for parent_y in variable_parents:
+            if full_graph_.has_edge(X, Y):
+                full_graph_.remove_edge(X, Y)
+            if parent_y in latent_variables:
+                full_graph_.add_edge(scaling_indicators[parent_y], dependent_var)
+
+        return full_graph_, dependent_var
+
+    def _identify(self, causal_graph):
+        exposure = causal_graph.get_role("exposures")[0]
+        outcome = causal_graph.get_role("outcomes")[0]
+        observed = causal_graph.get_role("observed")
+
+        latent_variables = causal_graph.get_role("latents")
+        scaling_indicators = self._get_scaling_indicators(causal_graph)
+
+        if (exposure in scaling_indicators.keys()) and (
+            scaling_indicators[exposure] == outcome
+        ):
+            logger.warning(
+                f"{outcome} is the scaling indicator of {exposure}. Please specify `scaling_indicators`"
+            )
```

Let's throw an error here asking user to specify the correct `scaling_indicators`.

---

Done